File: service/tests/service/seed.rs
use postgres::types::ToSql;

/// Trait that represents a type that can be seeded into the database
pub trait Seedable: std::fmt::Debug + Send + Sync {
    /// Generate the SQL needed to insert the seeded record into the database
    ///
    /// # Returns
    /// The SQL
    fn sql(&self) -> &str;

    /// Generate the binds needed to insert the seeded record into the database
    ///
    /// # Returns
    /// The binds
    fn binds(&self) -> Vec<&(dyn ToSql + Sync)>;
}

File: e2e/cypress/data/seed.js
/**
 * Base class that all Seed Data can inherit from
 */
export class SeedData {
  /**
   * Generate the SQL to insert the data into the database
   */
  async sql() {
    return "";
  }

  /**
   * Generate the binds for the SQL
   */
  async binds() {
    return [];
  }
}
